src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CfEnvConfigurer.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.cloudfoundry;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.loader.archive.JarFileArchive;
+import org.springframework.core.io.Resource;
+import org.springframework.util.StringUtils;
+
+public class CfEnvConfigurer {
+	public static final String SPRING_PROFILES_ACTIVE = "SPRING_PROFILES_ACTIVE";
+
+	public static final String SPRING_PROFILES_ACTIVE_FQN = "spring.profiles.active";
+
+	public static final String CLOUD_PROFILE_NAME = "cloud";
+	public static final String JBP_CONFIG_SPRING_AUTO_RECONFIGURATION = "JBP_CONFIG_SPRING_AUTO_RECONFIGURATION";
+	public static final String ENABLED_FALSE = "{ enabled: false }";
+
+	private static Log logger = LogFactory.getLog(CfEnvConfigurer.class);
+
+	public static Map<String,String> configureEnvironmentForCfEnv(Resource resource, Map<String, String> environment) {
+		try {
+			String scheme = resource.getURI().getScheme().toLowerCase();
+			if (scheme.equals("docker")) {
+				return environment;
+			}
+		}
+		catch (IOException e) {
+			throw new IllegalArgumentException(e.getMessage(), e);
+		}
+
+		Map<String, String> updatedEnvironment = new HashMap<>(environment);
+
+		if (CfEnvResolver.hasCfEnv(resource)) {
+			updatedEnvironment.putIfAbsent(JBP_CONFIG_SPRING_AUTO_RECONFIGURATION, ENABLED_FALSE);
+			if (appendToExistingEntry(updatedEnvironment, SPRING_PROFILES_ACTIVE, CLOUD_PROFILE_NAME)) {
```

Comment:
I wonder if we need to check all possible `RelaxedNames` when updating this entry.

Reply:
This is checking existing environment variables. I don't think I've ever seen env like springProfilesActive or spring-profiles-active.  If a user does this, they're on their own AFAIC.

Reply:
It's also possible to override the spring profiles with properties that go to SAJ, or command line args in task apps.  Maybe a better way to update the profile is is at the highest level of precedence instead of env vars.